deepparse/parser/address_parser.py

```diff
@@ -1044,61 +1034,33 @@ def _model_factory(
             # Set to default cache_path value
             cache_dir = CACHE_PATH
 
-        if "fasttext" in self.model_type:
-            if "fasttext-light" in self.model_type:
-                file_name = download_fasttext_magnitude_embeddings(
-                    cache_dir=cache_dir, verbose=verbose, offline=offline
-                )
+        model_factory = ModelFactory()
+        self.model = model_factory.create(
+            model_type=self.model_type,
+            cache_dir=cache_dir,
+            device=self.device,
+            output_size=prediction_layer_len,
+            attention_mechanism=attention_mechanism,
+            path_to_retrained_model=path_to_retrained_model,
+            offline=offline,
+            verbose=verbose,
+            **seq2seq_kwargs,
+        )
```

Comment:
```suggestion
        self.model = ModelFactory().create(
            model_type=self.model_type,
            cache_dir=cache_dir,
            device=self.device,
            output_size=prediction_layer_len,
            attention_mechanism=attention_mechanism,
            path_to_retrained_model=path_to_retrained_model,
            offline=offline,
            verbose=verbose,
            **seq2seq_kwargs,
        )
```

Reply:
I don't think we need to keep the factory as a variable object.

Reply:
Also, let's change the method name. Maybe something like _model_setup or else.

Reply:
Done!